From 8e50a6a1c5934ee897fc8fbc4ce08834a4848ea6 Mon Sep 17 00:00:00 2001
From: Song Zhao <song.zhao@amlogic.com>
Date: Sun, 26 Apr 2020 23:33:08 -0700
Subject: [PATCH] v4l2: return coded w/h in S_FMT

PD#SWPL-23981

Problem:
Application will use the w/h returned by S_FMT for resolution change.

Solution:
Return codec w/h in S_FMT

Verify:
U212 Cobalt
---
 drivers/amvdec_ports/aml_vcodec_dec.c     | 5 ++++-
 drivers/frame_provider/decoder/vp9/vvp9.c | 2 +-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/amvdec_ports/aml_vcodec_dec.c b/drivers/amvdec_ports/aml_vcodec_dec.c
index 7404bc93..d632cca0 100644
--- a/drivers/amvdec_ports/aml_vcodec_dec.c
+++ b/drivers/amvdec_ports/aml_vcodec_dec.c
@@ -1739,8 +1739,11 @@ static int vidioc_vdec_s_fmt(struct file *file, void *priv,
 		mutex_unlock(&ctx->state_lock);
 	}
 
-	if (!V4L2_TYPE_IS_OUTPUT(f->type))
+	if (!V4L2_TYPE_IS_OUTPUT(f->type)) {
 		ctx->cap_pix_fmt = pix_mp->pixelformat;
+		pix_mp->width = q_data->coded_width;
+		pix_mp->height = q_data->coded_height;
+	}
 
 	return 0;
 }
diff --git a/drivers/frame_provider/decoder/vp9/vvp9.c b/drivers/frame_provider/decoder/vp9/vvp9.c
index d817bafd..57233e00 100644
--- a/drivers/frame_provider/decoder/vp9/vvp9.c
+++ b/drivers/frame_provider/decoder/vp9/vvp9.c
@@ -10066,7 +10066,7 @@ static unsigned long run_ready(struct vdec_s *vdec, unsigned long mask)
 
 		if (ctx->param_sets_from_ucode) {
 			if (pbi->v4l_params_parsed) {
-				if ((ctx->cap_pool.in < ctx->dpb_size) &&
+				if ((ctx->cap_pool.in < pbi->used_buf_num) &&
 				v4l2_m2m_num_dst_bufs_ready(ctx->m2m_ctx) <
 				run_ready_min_buf_num)
 					ret = 0;
-- 
2.24.1

